snappy/kernel_os.go

```diff
@@ -89,26 +97,22 @@ func extractKernelAssets(s *snap.Info, snapf snap.File, flags InstallFlags, inte
 	}
 	defer dir.Close()
 
-	for _, src := range []string{s.Legacy.Kernel, s.Legacy.Initrd} {
-		if src == "" {
-			continue
-		}
-		if err := snapf.Unpack(src, dstDir); err != nil {
-			return err
-		}
-		src = filepath.Join(dstDir, src)
-		dst := filepath.Join(dstDir, dropVersionSuffix(src))
-		if err := os.Rename(src, dst); err != nil {
+	for _, src := range []string{
+		filepath.Join(s.MountDir(), "vmlinuz"),
```

Comment:
if I understood the discussion yesterday and look at the doc it seems this one now is named kernel.img